tests/conftest.py

```diff
@@ -41,14 +49,37 @@ def pytest_sessionstart(session):
     info = _get_info(redis_url)
     version = info["redis_version"]
     arch_bits = info["arch_bits"]
+    cluster_enabled = info["cluster_enabled"]
     REDIS_INFO["version"] = version
     REDIS_INFO["arch_bits"] = arch_bits
+    REDIS_INFO["cluster_enabled"] = cluster_enabled
 
     # module info
     redismod_url = session.config.getoption("--redismod-url")
     info = _get_info(redismod_url)
     REDIS_INFO["modules"] = info["modules"]
 
+    if cluster_enabled:
+        cluster_nodes = session.config.getoption("--cluster-nodes")
+        wait_for_cluster_creation(redis_url, cluster_nodes)
+
+
+def wait_for_cluster_creation(redis_url, cluster_nodes, timeout=20):
+    now = time.time()
+    timeout = now + timeout
+    print("Waiting for {0} cluster nodes to become available".
+          format(cluster_nodes))
+    while now < timeout:
+        try:
+            client = redis.RedisCluster.from_url(redis_url)
+            if len(client.get_nodes()) == cluster_nodes:
+                print("All nodes are available!")
+                break
+        except RedisClusterException:
+            pass
+        time.sleep(1)
```

Comment:
At some point, IMHO you probably want to raise a RedisClusterException if things don't't come up and we've made it all the way through the while loop.